### PR TITLE
feat: A infinite scroll should be present until the last post

### PR DESCRIPTION
**Pull request by motuncoded**
I made the posts to render in infinite scrolling until the last page.

### CODE REVIEW

File: pages/Posts.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    body: string;
+    userId: number;
+  };
+
+  type Error = {
+    message: string;
+  };
+  type Page = {
+    page: number;
+  };
```
Comment (Asylcreek) [inline pages/Posts.tsx]: Why not `type Page = number`?

File: pages/Posts.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      setError(error.message);
+    } finally {
+      setLoading(false);
+    }
+  };
+  useEffect(() => {
+    setTimeout(() => {
+      fetchPosts();
+    }, 1000);
+  }, [page]);
```
Comment (Asylcreek) [inline pages/Posts.tsx:48]: Why?

File: pages/Posts.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  };
+  type Page = {
+    page: number;
+  };
+
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<Error | null>(null);
+  const [page, setPage] = useState<Page>({ page: 1 });
+  const hasMore = page.page * 1 < 100;
```
Comment (Asylcreek) [inline pages/Posts.tsx]: This is not a good way to find out if the server's response has more pages.

**Review by Asylcreek — CHANGES_REQUESTED** [review]
Was this PR to show me how you'd like to implement the infinite scroll? Because your code doesn't show a working infinite scroll.

And it would be nice for you to deploy this so that I can see what it actually looks like.

File: pages/Posts.tsx (hunk 2 above)
Comment (motuncoded) [reply pages/Posts.tsx:48]: To schedule the post to render at 1 second

**Review by Asylcreek — APPROVED** [review]
I'm approving this so I can see it in action

**Comment by motuncoded** [comment]
Ok

**Pull request merged**